pkg/discovery/worker/compliance/affinity_processor.go

```diff
@@ -83,14 +83,28 @@ func (am *AffinityProcessor) processAffinityPerPod(pod *api.Pod, podsNodesMap ma
 		return
 	}
 
+	hostNode := findPodsNode(pod, podsNodesMap)
 	for _, node := range am.nodes {
+
 		if matchesNodeSelector(pod, node) && matchesNodeAffinity(pod, node) {
-			am.addAffinityAccessCommodities(pod, node, nodeAffinityAccessCommoditiesSold, nodeAffinityAccessCommoditiesBought)
+			am.addAffinityAccessCommodities(pod, node, hostNode, nodeAffinityAccessCommoditiesSold, nodeAffinityAccessCommoditiesBought)
 		}
 		if interPodAffinityMatches(pod, node, podsNodesMap) {
-			am.addAffinityAccessCommodities(pod, node, podAffinityCommodityDTOsSold, podAffinityCommodityDTOsBought)
+			am.addAffinityAccessCommodities(pod, node, hostNode, podAffinityCommodityDTOsSold, podAffinityCommodityDTOsBought)
+		}
+	}
+}
+
+// findPodsNode is needed because we do not get the maps key as a single unique value consistently.
+// The key which is a pointer when passed around functions can result in a new pointer variable
+// pointing to the same value (*pod.Pod).
+func findPodsNode(pod *api.Pod, podsNodesMap map[*api.Pod]*api.Node) *api.Node {
```

Comment:
@irfanurrehman I am not sure I fully understand. Where in our code that you see this pointer key might get changed? 

https://stackoverflow.com/questions/34585018/is-it-safe-to-use-pointers-as-map-keys-in-go